core/src/main/scala/chisel3/Mem.scala

```diff
@@ -74,28 +83,45 @@ sealed abstract class MemBase[T <: Data](val t: T, val length: BigInt) extends H
     * See the class documentation of the memory for more detailed information.
     */
   def apply(x: UInt): T = macro SourceInfoTransform.xArg
+  def apply(x: UInt, y: Clock): T = macro SourceInfoTransform.xyArg
```

Comment:
This is sort of an undocumented convention, but can you put this apply method above the `do_apply` it expands to? That makes it much easier to find the implementation of the method.